Code/Tools/AssetProcessor/native/ui/MainWindow.cpp

```diff
@@ -864,6 +876,79 @@ void MainWindow::SaveLogPanelState()
     }
 }
 
+void MainWindow::ShowOutgoingProductDependenciesContextMenu(const QPoint& pos)
+{
+    using namespace AssetProcessor;
+    auto productAt = [this](const QPoint& pos)
+    {
+        const QModelIndex assetIndex = ui->productAssetDetailsPanel->GetOutgoingProductDependenciesTreeView()->indexAt(pos);
+        return static_cast<ProductDependencyTreeItem*>(assetIndex.internalPointer());
+    };
+    const ProductDependencyTreeItem* cachedAsset = productAt(pos);
+
+    if (!cachedAsset || !cachedAsset->GetData())
+    {
+        return;
+    }
+    AZStd::string productName = cachedAsset->GetData()->m_productName;
+
+    QMenu menu(this);
+    menu.setToolTipsVisible(true);
+    QAction* productAction = menu.addAction(
+        tr("Go to product asset"), this,
+        [&, productName]()
+        {
+            ui->sourceAssetDetailsPanel->GoToProduct(productName);
+        });
+    if (productName.empty())
+    {
+        productAction->setDisabled(true);
+        productAction->setToolTip(tr("This asset is currently selected."));
+    }
+    else
+    {
+        productAction->setToolTip(tr("Selects this asset."));
+    }
+    menu.exec(ui->productAssetDetailsPanel->GetOutgoingProductDependenciesTreeView()->viewport()->mapToGlobal(pos));
+}
+
+void MainWindow::ShowIncomingProductDependenciesContextMenu(const QPoint& pos)
+{
+    using namespace AssetProcessor;
+    auto productAt = [this](const QPoint& pos)
+    {
+        const QModelIndex assetIndex = ui->productAssetDetailsPanel->GetIncomingProductDependenciesTreeView()->indexAt(pos);
+        return static_cast<ProductDependencyTreeItem*>(assetIndex.internalPointer());
+    };
+    const ProductDependencyTreeItem* cachedAsset = productAt(pos);
```

Comment:
I am trying to understand the value of creating a lambda function to call two lines of code... is the productAt() suppose to be used latter on?

Reply:
Seems not. This code snippet is ported from previous PR #6487 though.

Reply:
removed lambdas and made a helper function in MainWindow that can get the TreeItem of incoming and outgoing dependency TreeViews.